core/src/main/java/com/datastrato/gravitino/authorization/AdminManager.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.authorization;
+
+import com.datastrato.gravitino.Config;
+import com.datastrato.gravitino.Configs;
+import com.datastrato.gravitino.Entity;
+import com.datastrato.gravitino.EntityAlreadyExistsException;
+import com.datastrato.gravitino.EntityStore;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.exceptions.UserAlreadyExistsException;
+import com.datastrato.gravitino.meta.AuditInfo;
+import com.datastrato.gravitino.meta.UserEntity;
+import com.datastrato.gravitino.storage.IdGenerator;
+import com.datastrato.gravitino.utils.EntitySpecificConstants;
+import com.datastrato.gravitino.utils.PrincipalUtils;
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * There are two kinds of admin roles in the system: service admin and metalake admin. The service
+ * admin is configured instead of managing by APIs. It is responsible for creating metalake admin.
+ * If Gravitino enables authorization, service admin is required. Metalake admin can create a
+ * metalake or drops its metalake. The metalake admin will be responsible for managing the access
+ * control. AdminManager operates underlying store using the lock because kv storage needs the lock.
+ */
+public class AdminManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AdminManager.class);
+
+  private final EntityStore store;
+  private final IdGenerator idGenerator;
+  private final List<String> admins;
+
+  public AdminManager(EntityStore store, IdGenerator idGenerator, Config config) {
+    this.store = store;
+    this.idGenerator = idGenerator;
+    this.admins = config.get(Configs.SERVICE_ADMINS);
```

Comment:
Please change to `serviceAdmin` to distinguish with the metalakeAdmin.

Reply:
OK.